include/spirv-tools/libspirv.hpp

```diff
@@ -278,7 +278,7 @@ class FuzzerOptions {
 // provides methods for assembling, disassembling, and validating.
 //
 // Instances of this class provide basic thread-safety guarantee.
-class SpirvTools {
+class SPIRV_TOOLS_EXPORT SpirvTools {
```

Comment:
Does struct Impl at line 369 need to have it too?
VS 2017 is erroring out.

Reply:
I have a feeling like the warning is wrong (the failure is actually a -Werror warning escalation). The whole idea of pimpl pattern is to hide the implementation from the users. And SpirvTools has no inline or auto-generated members, so there should be no way how the user might access `impl_` member from his/her translation unit.

Reply:
The change doesn't build.

E.g.

libspirv.hpp(370): warning C4251: 'spvtools::SpirvTools::impl_': class 'std::unique_ptr<spvtools::SpirvTools::Impl,std::default_delete<_Ty>>' needs to have dll-interface to be used by clients of class 'spvtools::SpirvTools'


So the change can't land as it is.  We do elevate warnings to errors, so let's fix them. The compiler is our partner even when it's not as smart as we think it should be.

Reply:
Hi, @dneto0!

It depends on your policy on exported symbols and binary compatibility. Do you want the private members to be exported in the final DLL? 

If yes, I can just add `SPIRV_TOOLS_EXPORT` to the private member. If no, I can silence the warning using pragmas (though I will probably need some help with that). 

Here is Krita we usually mark the private members as `KRITAUI_NO_EXPORT` to make sure they are not exported.

Reply:
Hi, @dneto0!

I have pushed a commit that explicitly disables the warning. Please let me know if it is not what the policy expects.